### PR TITLE
Test book controller

### PR DESCRIPTION
**Pull request by Sspadea**
## Summary

- testing book controller
- shared_context authenticated_user

## Trello Card

https://trello.com/c/emwPLdT8/21-2-crear-tests-para-rutas-index-y-show-de-book

## Screenshots or Screencasts

Captura de Pantalla 2021-06-22 a la(s) 13 06 34

### CODE REVIEW

File: app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      # before_action :authenticate_user!
+      include Wor::Paginate
+
+      def index
```
Comment (blacksam07) [inline app/controllers/api/v1/books_controller.rb:4]: if very different to #11

**Review by blacksam07 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        books = Rack::Reducer.call(params, dataset: Book.all, filters: [
+          ->(author:) { where('lower(author) like ?', "%#{author.downcase}%") },
+          ->(genre:) { where('lower(book_genre) like ?', "%#{genre.downcase}%") },
+          ->(title:) { where('lower(title) like ?', "%#{title.downcase}%") }
+        ])
```
Comment (redwarewolf) [inline app/controllers/api/v1/books_controller.rb]: This logic shouldn't be here but on a service.

File: app/controllers/api/v1/books_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        render_paginated books, each_serializer: BookSerializer
+      end
+
+      def show
+        book = Book.friendly.find(params[:id])
```
Comment (redwarewolf) [inline app/controllers/api/v1/books_controller.rb:10]: same here

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when fetching a book' do
```
Comment (redwarewolf) [inline spec/controllers/api/v1/books_controller_spec.rb:25]: And the bad scenario?

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+require './spec/shared_context/authenticated_user'
+
+describe Api::V1::BooksController, type: :controller do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'when fetching all the books' do
```
Comment (redwarewolf) [inline spec/controllers/api/v1/books_controller_spec.rb:8]: and the bad scenario?

**Review by redwarewolf — CHANGES_REQUESTED** [review]

**Review by redwarewolf — APPROVED** [review]

**Review by blacksam07 — APPROVED** [review]

**Pull request merged**